backend/routes/stocks.js

```diff
@@ -21,18 +21,22 @@ router.get("/stocks/:ticker", async (req, res) => {
         where: { ticker }
     });
 
-    const user = req.session.user;
+    if (stock) {
```

Comment:
stock is not a boolean. It is better to explicitly check for it not being undefined